src/modules/cpu/kernel/noise_gaussian.hpp

```diff
@@ -1390,3 +1390,752 @@ RppStatus gaussian_noise_i8_i8_host_tensor(Rpp8s *srcPtr,
+        Rpp32u offset = batchCount * srcGenericDescPtr->strides[0];
+        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = (bufferLength / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+        RpptXorwowStateBoxMuller xorwowState;
+#if __AVX2__
+        __m256i pxXorwowStateX[5], pxXorwowStateCounter;
+        rpp_host_rng_xorwow_state_offsetted_avx(xorwowInitialStatePtr, xorwowState, offset, pxXorwowStateX, &pxXorwowStateCounter);
+
+        __m256 pGaussianNoiseParams[2];
+        pGaussianNoiseParams[0] = _mm256_set1_ps(mean);
+        pGaussianNoiseParams[1] = _mm256_set1_ps(stdDev);
+#else
+        __m128i pxXorwowStateX[5], pxXorwowStateCounter;
+        rpp_host_rng_xorwow_state_offsetted_sse(xorwowInitialStatePtr, xorwowState, offset, pxXorwowStateX, &pxXorwowStateCounter);
+
+        __m128 pGaussianNoiseParams[2];
+        pGaussianNoiseParams[0] = _mm_set1_ps(mean);
+        pGaussianNoiseParams[1] = _mm_set1_ps(stdDev);
+#endif
+        // Gaussian Noise without fused output-layout toggle (NDHWC -> NDHWC)
+        if((srcGenericDescPtr->dims[4] == 3) && (srcGenericDescPtr->layout == RpptLayout::NDHWC) && (dstGenericDescPtr->layout == RpptLayout::NDHWC))
+        {
+            srcPtrChannel = srcPtrImage + (roi.xyzwhdROI.xyz.z * srcGenericDescPtr->strides[2]) + (roi.xyzwhdROI.xyz.y * srcGenericDescPtr->strides[3]) + (roi.xyzwhdROI.xyz.x * layoutParams.bufferMultiplier);
+
+            Rpp8u *srcPtrDepth, *dstPtrDepth;
+            srcPtrDepth = srcPtrChannel;
+            dstPtrDepth = dstPtrChannel;
+            for(int i = 0; i < roi.xyzwhdROI.roiDepth; i++)
+            {
+                Rpp8u *srcPtrRow, *dstPtrRow;
+                srcPtrRow = srcPtrDepth;
+                dstPtrRow = dstPtrDepth;
+                for(int j = 0; j < roi.xyzwhdROI.roiHeight; j++)
+                {
+                    Rpp8u *srcPtrTemp, *dstPtrTemp;
+                    srcPtrTemp = srcPtrRow;
+                    dstPtrTemp = dstPtrRow;
+
+                    int vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                    {
+#if __AVX2__
+                        __m256 p[6];
+                        rpp_simd_load(rpp_load48_u8pkd3_to_f32pln3_avx, srcPtrTemp, p);                                 // simd loads
+                        rpp_multiply48_constant(p, avx_p1op255);                                                        // u8 normalization to range[0,1]
+                        compute_gaussian_noise_48_host(p, pxXorwowStateX, &pxXorwowStateCounter, pGaussianNoiseParams); // gaussian_noise adjustment
+                        rpp_multiply48_constant(p, avx_p255);                                                           // u8 un-normalization
```

Comment:
Not sure if the adjustment for 255 is needed in a tensor. Lets discuss and come to consensus for this.